lib/i18n/backend/base.rb

```diff
@@ -95,10 +95,19 @@ def available_locales
       end
 
       def reload!
+        eager_load! if eager_loaded?
```

Comment:
Would not call `eager_load!` here cause the same problem that you were trying to avoid with our patch? Every time a new I18n path is added we would eager load the files.

I guess this will be fine because you would only call `eager_load` after the `load_path` are all set.

Reply:
Yeah it's a trade off. Either `reload!` becomes eager once `eager_load!` was called once, and you risk wasting compute in reloading the same data multiple times if it's called afterwards.

Or either you don't and then you risk "disabling" eager loading if it's called later.

I think I prefer the former.